### PR TITLE
OptionVals and no requires() in materializer

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/akka/akka/issues/22438
Resolves https://github.com/akka/akka/issues/22437

Measured just on my laptop to numbers are a bit nonsense, but here they are for reference.
Too much other stuff happening on my laptop to get a meaningful measurement here I think.

Before:

```
[info]
[info] Benchmark                               (complexity)   Mode  Cnt       Score       Error  Units
[info] MaterializationBenchmark.flow_with_map             1  thrpt   20  173573.332 ± 24454.250  ops/s
[info] MaterializationBenchmark.flow_with_map            10  thrpt   20   94503.123 ± 11538.851  ops/s
```

After: 

```
[info] Benchmark                               (complexity)   Mode  Cnt       Score       Error  Units
[info] MaterializationBenchmark.flow_with_map             1  thrpt   20  169334.168 ± 19569.725  ops/s
[info] MaterializationBenchmark.flow_with_map            10  thrpt   20   85744.876 ±  8376.737  ops/s
```

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/ActorProcessor.scala [hunk 1]
```diff
@@ -47,8 +47,9 @@ private[akka] class ActorProcessor[I, O](impl: ActorRef) extends ActorPublisher[
  * INTERNAL API
  */
 private[akka] abstract class BatchingInputBuffer(val size: Int, val pump: Pump) extends DefaultInputTransferStates {
-  require(size > 0, "buffer size cannot be zero")
-  require((size & (size - 1)) == 0, "buffer size must be a power of two")
+  if (size < 1) throw new IllegalArgumentException(s"buffer size MSUT be positive (was: $size")
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/ActorProcessor.scala]: typo MSUT
(I see no reason for using capital letters)

File: akka-stream/src/main/scala/akka/stream/impl/ActorProcessor.scala [hunk 2]
```diff
@@ -47,8 +47,9 @@ private[akka] class ActorProcessor[I, O](impl: ActorRef) extends ActorPublisher[
...
+  if ((size & (size - 1)) != 0) throw new IllegalArgumentException("buffer size must be a power of two")
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/ActorProcessor.scala]: including `(was: $size")` here also would be nice

File: akka-stream/src/main/scala/akka/stream/impl/ActorProcessor.scala (hunk 1 above)
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/ActorProcessor.scala]: missing end `)` in the message
`(was: $size)"`

**Review by patriknw — CHANGES_REQUESTED** [review]
just some small things

**Review by patriknw — APPROVED** [review]
LGTM

**Review by johanandren — APPROVED** [review]
I approve!

**Pull request merged**